storage_drivers/fake/fake.go

```diff
@@ -50,6 +51,9 @@ type StorageDriver struct {
 
 	physicalPools map[string]*storage.Pool
 	virtualPools  map[string]*storage.Pool
+
+	// Snapshots saves info about Snapshots created on this driver
+	Snapshots map[string][]storage.Snapshot
```

Comment:
not sure if this makes a difference since delete is unclear, but would `map[string][string]storage.Snapshot` be better?